src/frameworks/compose/discover/runtime/node.ts

```diff
@@ -0,0 +1,216 @@
+import { readOrNull } from "../filesystem";
+import { FileSystem, FrameworkSpec, Runtime } from "../types";
+import { RuntimeSpec } from "../types";
+import { frameworkMatcher } from "../frameworkMatcher";
+import { LifecycleCommands } from "../types";
+import { Command } from "../types";
+import { FirebaseError } from "../../../../error";
+import { logger } from "../../../../../src/logger";
+
+export interface PackageJSON {
+  dependencies?: Record<string, string>;
+  devDependencies?: Record<string, string>;
+  scripts?: Record<string, string>;
+  engines?: Record<string, string>;
+}
+type PackageManager = "npm" | "yarn";
+
+const supportedNodeVersion = 18;
```

Comment:
nit* let's make this a string[] - we know it's going to be a list soon.

Reply:
Done